crates/nu-parser/src/parse_keywords.rs

```diff
@@ -3317,6 +3317,16 @@ pub fn parse_mut(working_set: &mut StateWorkingSet, spans: &[Span]) -> Pipeline
 
 pub fn parse_source(working_set: &mut StateWorkingSet, lite_command: &LiteCommand) -> Pipeline {
     let spans = &lite_command.parts;
+
+    if working_set.source_recursions > working_set.permanent_state.config.recursion_limit {
+        working_set.error(ParseError::RecursionLimitReached(
+            working_set.permanent_state.config.recursion_limit,
+            spans[0],
+        ));
+        return garbage_pipeline(spans);
+    }
+    working_set.source_recursions += 1;
```

Comment:
Am I reading this correctly that this would simply increase on every parse of `source` in this `StateWorkingSet`? So if you would have more than `config.recursion_limit` invocations of `source` independent of the file sourced, this would error out?

Reply:
Yes - that does mean, admittedly, that if files each `source`d another 50 times, this would throw an error. It seems that the other PR is attempting to find explicitly find recursive sources. I suppose this PR might be better characterised as "extending recursion limits to `source`s", with preventing overflows on circular `source`s a happy coincidence?